api/types/databaseserver.go

```diff
@@ -0,0 +1,391 @@
+	return LabelsAsString(s.Metadata.Labels, s.Spec.DynamicLabels)
+}
+
+// GetDescription returns the database server description.
+func (s *DatabaseServerV2) GetDescription() string {
+	return s.Spec.Description
+}
+
+// GetProtocol returns the database server protocol.
+func (s *DatabaseServerV2) GetProtocol() string {
+	return s.Spec.Protocol
+}
+
+// GetURI returns the database connection address.
+func (s *DatabaseServerV2) GetURI() string {
+	return s.Spec.URI
+}
+
+// GetCA returns the database CA certificate bytes.
+func (s *DatabaseServerV2) GetCA() []byte {
+	return s.Spec.CACert
+}
+
+// GetRegion returns the AWS region for RDS/Aurora databases.
+func (s *DatabaseServerV2) GetRegion() string {
+	return s.Spec.AWS.Region
+}
+
+// IsAWS returns true if this database represents AWS RDS/Aurora instance.
+func (s *DatabaseServerV2) IsAWS() bool {
+	return s.Spec.AWS.Region != ""
+}
+
+// String returns the server string representation.
+func (s *DatabaseServerV2) String() string {
+	return fmt.Sprintf("DatabaseServer(Name=%v, Version=%v, Labels=%v)",
+		s.GetName(), s.GetTeleportVersion(), s.GetStaticLabels())
+}
+
+// CheckAndSetDefaults checks and sets default values for any missing fields.
+func (s *DatabaseServerV2) CheckAndSetDefaults() error {
+	if err := s.Metadata.CheckAndSetDefaults(); err != nil {
+		return trace.Wrap(err)
+	}
+	if s.Kind == "" {
+		return trace.BadParameter("database server %q kind is empty", s.GetName())
+	}
+	for key := range s.Spec.DynamicLabels {
+		if !IsValidLabelKey(key) {
+			return trace.BadParameter("database server %q invalid label key: %q", s.GetName(), key)
+		}
+	}
+	if s.Spec.Protocol == "" {
+		return trace.BadParameter("database server %q protocol is empty", s.GetName())
+	}
+	if s.Spec.URI == "" {
+		return trace.BadParameter("database server %q URI is empty", s.GetName())
+	}
+	if s.Spec.Hostname == "" {
+		return trace.BadParameter("database server %q hostname is empty", s.GetName())
+	}
+	if s.Spec.HostID == "" {
+		return trace.BadParameter("database server %q host ID is empty", s.GetName())
+	}
+	return nil
+}
+
+// Copy returns a copy of this database server object.
+func (s *DatabaseServerV2) Copy() DatabaseServer {
+	return &DatabaseServerV2{
+		Kind:     KindDatabaseServer,
+		Version:  V2,
+		Metadata: s.Metadata,
+		Spec:     s.Spec,
+	}
+}
+
+// DatabaseServerSpecV2Schema is JSON schema for a database server spec.
+const DatabaseServerSpecV2Schema = `{
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "description": {"type": "string"},
+    "protocol": {"type": "string"},
+    "uri": {"type": "string"},
+    "ca_cert": {"type": "string"},
+    "aws": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "region": {"type": "string"}
+      }
+    },
+    "version": {"type": "string"},
+    "hostname": {"type": "string"},
+    "host_id": {"type": "string"},
+    "dynamic_labels": {
```

Comment:
Yeah, replied below to a similar comment, static labels are actually in metadata.labels since the DatabaseServer is a separate resource. And dynamic labels are a part of the spec.

Reply:
Is there any way to locate static/dynamic labels on the same level?

Reply:
I can move them to the spec as well if you feel it would be better, though I'm not sure the benefit - this is just the internal resource representation so I thought that using Metadata.Labels would be a good idea since they're, well, regular resource labels. In file configuration that users write, they're obviously on the same level for each "database" definition, but each "database" instance gets converted to DatabaseServer resource internally, static labels are set in metadata. Let me know what you think.

Reply:
thinking about this more, shouldn't the `database_service` evaluate dynamic labels and present the result as regular labels on the resource?
IOTW, should `static/dynamic_labels` distinction even show up in the backend?

Reply:
Well, "dynamic_labels" on the resource stores both the commands that are run periodically (and their intervals) and the evaluated values, so I don't think we can unify them under just regular labels.